src/reports.cpp

```diff
@@ -934,7 +934,10 @@ static int attack_info(reports::context & rc, const attack_type &at, config &res
 	}
 
 	{
-		auto ctx = at.specials_context_for_listing(u.side() == rc.screen().playing_side());
+		//If we have a second unit, do the 2-unit specials_context
+		auto ctx = (sec_u == nullptr) ? at.specials_context_for_listing(u.side() == rc.screen().playing_side()) :
+						at.specials_context(unit_const_ptr(&u), unit_const_ptr(sec_u), hex, sec_u->get_location(), u.side() == rc.screen().playing_side(), sec_u_weapon);
```

Comment:
The check for "attacking" is duplicated across these two lines. Could you extract it into a named variable?

Reply:
Done.